frontend/src/features/map_planning/components/BaseStage.tsx

```diff
@@ -156,29 +166,105 @@ export const BaseStage = ({
     }
   };
 
-  // Event listener responsible for adding a single shape to the transformer
-  const addToTransformer = (node: Shape<ShapeConfig>) => {
-    const nodes = trRef.current?.getNodes() || [];
-    if (!nodes.includes(node)) {
-      trRef.current?.nodes([node]);
-    }
-  };
-
-  // Add event listeners to all shapes
-  // This will trigger on every rerender, maybe this could be improved?
-  stageRef.current?.children
-    ?.flatMap((layer) => layer.children)
-    .filter((shape) => shape?.name() !== 'selectionRect' && !shape?.name().includes('transformer'))
-    .forEach((shape) => {
-      if (!shape?.eventListeners['click']) {
-        shape?.addEventListener('click', () => {
-          addToTransformer(shape as Shape<ShapeConfig>);
-        });
-      }
+  const dispatchUpdate = (
+    nodes: Node<NodeConfig>[],
+    type: 'OBJECT_UPDATE_POSITION' | 'OBJECT_UPDATE_TRANSFORM',
+  ) => {
+    dispatch({
+      type,
+      payload: nodes.map((o) => ({
+        id: o.id(),
+        type: o.attrs.type,
+        index: o.attrs.index,
+        height: o.height(),
+        width: o.width(),
+        x: o.x(),
+        y: o.y(),
+        rotation: o.rotation(),
+        scaleX: o.scaleX(),
+        scaleY: o.scaleY(),
+      })),
     });
+  };
 
   return (
     <div className="h-screen w-screen overflow-hidden">
+      <div className="absolute z-10 flex h-10 items-center gap-2 pl-2 pt-12">
+        <SimpleButton
+          className="w-32"
+          onClick={() =>
+            dispatch({
+              type: 'OBJECT_ADD',
+              payload: {
+                index: 'plant',
+                id: Math.random().toString(36).slice(2, 9),
```

Comment:
Is this save enough in terms of uniqueness. Are there any consequences when two object have the same id?

Reply:
@absurd-turtle This is example code, that shows how an object should be added to the store. 
I will add a comment that this is still a TODO but it should be handled in another issue.

Reply:
Ah ok, that makes sense